File: example/go/config/Ai/Ai_action.go
package ai

type Ai_action struct {
    iD int
    desc string //描述
    formulaID int //公式
    argIList []int //参数(int)1
    argSList []int //参数(string)1
}

//getters
func (t *Ai_action) GetID() int {
    return t.iD
}

func (t *Ai_action) GetDesc() string {
    return t.desc
}

func (t *Ai_action) GetFormulaID() int {
    return t.formulaID
}

func (t *Ai_action) GetArgIList() []int {
    return t.argIList
}

func (t *Ai_action) GetArgSList() []int {
    return t.argSList
}

var all []Ai_action
func GetAll() []Ai_action {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Ai_action
func Get(key int) (Ai_action,bool){
    return allMap[key]
}
        public static Ai_action Get(int iD)
        {
            Ai_action v;
            return all.TryGetValue(iD, out v) ? v : null;
        }

        public static List<Ai_action> All()
        {
            return all.OrderedValues;
        }

        public static List<Ai_action> Filter(Predicate<Ai_action> predicate)
        {
            var r = new List<Ai_action>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Ai_action>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.ID, self);
            }
        }

File: example/java/config/ai/AiAction.java
package config.ai;

// 来自：ai_行为/ai行为.xlsx[AI_ACTION_中文测试], ai_行为/ai行为.xlsx[AI_ACTION_1_继续测试]
public class AiAction {
    private int iD;
    private String desc;
    private int formulaID;
    private java.util.List<Integer> argIList;
    private java.util.List<Integer> argSList;

    private AiAction() {
    }

    public static AiAction _create(configgen.genjava.ConfigInput input) {
        AiAction self = new AiAction();
        self.iD = input.readInt();
        self.desc = input.readStringInPool();
        self.formulaID = input.readInt();
        {
            int c = input.readInt();
            if (c == 0) {
                self.argIList = java.util.Collections.emptyList();
            } else {
                self.argIList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.argIList.add(input.readInt());
                }
            }
        }
        {
            int c = input.readInt();
            if (c == 0) {
                self.argSList = java.util.Collections.emptyList();
            } else {
                self.argSList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.argSList.add(input.readInt());
                }
            }
        }
        return self;
    }

    public int getID() {
        return iD;
    }

    /**
     * 描述
     */
    public String getDesc() {
        return desc;
    }

    /**
     * 公式
     */
    public int getFormulaID() {
        return formulaID;
    }

    /**
     * 参数(int)1
     */
    public java.util.List<Integer> getArgIList() {
        return argIList;
    }

    /**
     * 参数(string)1
     */
    public java.util.List<Integer> getArgSList() {
        return argSList;
    }

    @Override
    public String toString() {
        return "(" + iD + "," + desc + "," + formulaID + "," + argIList + "," + argSList + ")";
    }

    public static AiAction get(int iD) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getAiAiAction(iD);
    }

    public static java.util.Collection<AiAction> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allAiAiAction();
    }
    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            int c = input.readInt();
            mgr.ai_ai_action_All = new java.util.LinkedHashMap<>(c);
            for (; c > 0; c--) {
                AiAction self = AiAction._create(input);
                mgr.ai_ai_action_All.put(self.iD, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            // no resolve
        }

    }

}
